magicctapipe/io/io.py

```diff
@@ -0,0 +1,941 @@
+        ).copy()
+
+        df_events["multiplicity"] = df_events.groupby(group_index).size()
+        df_events.query(f"multiplicity == {multiplicity}", inplace=True)
+
+        n_events = int(len(df_events.groupby(group_index).size()))
+        percentage = np.round(100 * n_events / n_events_total, 1)
+
+        logger.info(
+            f"\t{tel_combo} (type {combo_type}): {n_events} events ({percentage}%)"
+        )
+
+        event_data_stereo.loc[df_events.index, "combo_type"] = combo_type
+
+    return event_data_stereo
+
+
+def get_dl2_mean(event_data, weight_type="simple", group_index=["obs_id", "event_id"]):
+    """
+    Gets the mean DL2 parameters per shower event.
+
+    Parameters
+    ----------
+    event_data: pandas.core.frame.DataFrame
+        Data frame of shower events
+    weight_type: str
+        Type of the weights for the telescope-wise DL2 parameters -
+        "simple" does not use any weights for calculations,
+        "variance" uses the inverse of the RF variance, and
+        "intensity" uses the linear-scale intensity parameter
+    group_index: list
+        Index to group telescope-wise events
+
+    Returns
+    -------
+    event_data_mean: pandas.core.frame.DataFrame
+        Data frame of the shower events with the mean DL2 parameters
+
+    Raises
+    ------
+    ValueError
+        If the input weight type is not known
+    """
+
+    is_simulation = "true_energy" in event_data.columns
+
+    # Create a mean data frame
+    if is_simulation:
+        params = ["combo_type", "multiplicity", "true_energy", "true_alt", "true_az"]
+    else:
+        params = ["combo_type", "multiplicity", "timestamp"]
+
+    event_data_mean = event_data[params].groupby(group_index).mean()
+
+    # Calculate the mean pointing direction
+    pointing_az_mean, pointing_alt_mean = calculate_mean_direction(
+        lon=event_data["pointing_az"], lat=event_data["pointing_alt"], unit="rad"
+    )
+
+    event_data_mean["pointing_alt"] = pointing_alt_mean
+    event_data_mean["pointing_az"] = pointing_az_mean
+
+    # Define the weights for the DL2 parameters
+    if weight_type == "simple":
+        energy_weights = 1
+        direction_weights = None
+        gammaness_weights = 1
+
+    elif weight_type == "variance":
+        energy_weights = 1 / event_data["reco_energy_var"]
+        direction_weights = 1 / event_data["reco_disp_var"]
+        gammaness_weights = 1 / event_data["gammaness_var"]
+
+    elif weight_type == "intensity":
+        energy_weights = event_data["intensity"]
+        direction_weights = event_data["intensity"]
+        gammaness_weights = event_data["intensity"]
+
+    else:
+        raise ValueError(f"Unknown weight type '{weight_type}'.")
+
+    df_events = pd.DataFrame(
+        data={
+            "energy_weight": energy_weights,
+            "gammaness_weight": gammaness_weights,
+            "weighted_energy": np.log10(event_data["reco_energy"]) * energy_weights,
```

Comment:
weighted_log_energy (??)

Reply:
Thanks, I changed the parameter name with "log".